server/server.go

```diff
@@ -72,6 +72,7 @@ type Server struct {
 	// a supervisor automatically restart it, then new client connection will be created, but we can't server it.
 	// So we just stop the listener and store to force clients to chose other TiDB servers.
 	stopListenerCh chan struct{}
+	ppDecoder      *proxyProtocolDecoder
```

Comment:
`pp` in `ppDecoder` is not an obvious naming. Consider something like `proxyProtocolDecoder`.